crypto/keystack.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+/**
+ * Provides the {@linkcode KeyStack} class which implements the
+ * {@linkcode KeyRing} interface for managing rotatable keys.
+ *
+ * @module
+ */
+
+import * as base64 from "../encoding/base64.ts";
+
+/** Types of data that can be signed cryptographically. */
+export type Data = string | number[] | ArrayBuffer | Uint8Array;
+
+/** Types of keys that can be used to sign data. */
+export type Key = string | number[] | ArrayBuffer | Uint8Array;
+
+/** An abstract interface for a keyring which handles signing of data based on
+ * a string based digest. */
+export interface KeyRing {
```

Comment:
Why does this have a different name from the class name? How about naming both KeyStack?

Reply:
As stated, it is an abstract interface of which `KeyStack` is _an_ implementation of it. What we are establishing here though is a generic interface people would then be able to produce their own implementations if they wanted to.  There isn't _yet_ a good home for this interface outside of here at the moment.

Reply:
Ok. Sounds fair enough